editor/src/clj/editor/model.clj

```diff
@@ -137,43 +166,254 @@
+                                                          (g/->error _node-id :materials :warning material
+                                                                     (format "'%s' is not defined in the mesh. Clear the field to delete it."
+                                                                             name)))
+                                                        (prop-resource-error :fatal _node-id :materials material "Material"))
+                                               :prop-kw :material
+                                               :edit-type {:type resource/Resource
+                                                           :ext "material"
+                                                           :clear-fn (fn [_ _]
+                                                                       (remove-material model-node-id name))}}
+                                              should-be-deleted
+                                              (assoc :original-value fake-resource))]]
+                                    (map (fn [{:keys [_node-id sampler-name image-resource]}]
+                                           [(keyword (str "__sampler__" sampler-name "__" name))
+                                            {:node-id _node-id
+                                             :label sampler-name
+                                             :type resource/Resource
+                                             :value image-resource
+                                             :prop-kw :image-resource
+                                             :edit-type {:type resource/Resource
+                                                         :ext (conj image/exts "cubemap")}}]))
+                                    texture-binding-infos))))
+                        material-binding-infos)
+                      ;; required materials that are not defined:
+                      (into
+                        (comp
+                          (remove proto-ids)
+                          (map (fn [id-to-add]
+                                 [(keyword (str "__material__" id-to-add))
+                                  {:node-id _node-id
+                                   :label id-to-add
+                                   :value nil
+                                   :type resource/Resource
+                                   :error (prop-resource-error :fatal _node-id :material nil "Material")
+                                   :edit-type {:type resource/Resource
+                                               :ext "material"
+                                               :set-fn (fn [_evaluation-context id _old new]
+                                                         (add-material id id-to-add new))}}])))
+                        mesh-material-ids))]
+    (-> _declared-properties
+        (update :properties into new-props)
+        (update :display-order into (map first) new-props))))
+
```

Comment:
The order things appear in the Properties view seems somewhat arbitrary. Would it make sense to sort the materials and their respective textures by name?